src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorComponentTests.cs

```diff
@@ -830,6 +831,79 @@ public async Task LineMapping_Tabs()
+        result.VerifyOutputsMatchBaseline();
+
+        var original = project.AdditionalDocuments.Single();
+        var originalText = await original.GetTextAsync();
+        Assert.Equal(source, originalText.ToString());
+        var generated = result.GeneratedSources.Single();
+        var generatedText = generated.SourceText;
+        var generatedTextString = generatedText.ToString();
+
+        // Find the usings and verify their mapping.
+        var snippets = new[]
+        {
+            "using System.Net.Http",
+            "using static Microsoft.AspNetCore.Components.Web.RenderMode"
+        };
+        var expectedLines = new[] { 0, 1 };
+        var expectedSemicolonMappings = new[] { true, false };
+        var originalIndex = -1;
+        var generatedIndex = -1;
+        foreach (var (snippet, expectedLine, expectedSemicolonMapping) in snippets.Zip(expectedLines, expectedSemicolonMappings))
+        {
+            originalIndex = source.IndexOf(snippet, originalIndex + 1, StringComparison.Ordinal);
+            generatedIndex = generatedTextString.IndexOf(snippet, generatedIndex + 1, StringComparison.Ordinal);
+            var generatedSpan = new TextSpan(generatedIndex, snippet.Length);
+            Assert.Equal(snippet, generatedText.ToString(generatedSpan));
+            var mapped = generated.SyntaxTree.GetMappedLineSpan(generatedSpan);
+            Assert.True(mapped.IsValid);
+            Assert.True(mapped.HasMappedPath);
+            Assert.Equal("Shared/Component1.razor", mapped.Path);
+            Assert.Equal(expectedLine, mapped.StartLinePosition.Line);
+            Assert.Equal(expectedLine, mapped.EndLinePosition.Line);
+            var mappedSpan = originalText.Lines.GetTextSpan(mapped.Span);
+            Assert.Equal(snippet, originalText.ToString(mappedSpan));
+            Assert.Equal(new TextSpan(originalIndex, snippet.Length), mappedSpan);
+
+            // Find the using's semicolon.
+            // The first one is mapped back to the razor file, the second is not due to `#line hidden` emitted for it.
+            // The first one being mapped is not nice - the semicolon does not exist in the razor file,
+            // see https://github.com/dotnet/razor/issues/10375.
+            var usingDirective = generated.SyntaxTree.GetRoot().DescendantNodes().OfType<UsingDirectiveSyntax>()
+                .Single(s => s.Name.ToString() == snippet.Split(' ')[^1]);
+            var semicolon = usingDirective.SemicolonToken;
+            var mappedSemicolon = semicolon.GetLocation().GetMappedLineSpan();
+            Assert.True(mappedSemicolon.IsValid);
+            Assert.Equal(expectedSemicolonMapping, mappedSemicolon.HasMappedPath);
+            if (expectedSemicolonMapping)
+            {
+                Assert.Equal("Shared/Component1.razor", mappedSemicolon.Path);
+                Assert.Throws<ArgumentOutOfRangeException>(() => originalText.Lines.GetTextSpan(mappedSemicolon.Span));
+            }
```

Comment:
> > the newline at the end of a `@using` statement is no longer part of the using declaration itself
> 
> Sounds like maybe this fixes #10375 then?

Added this test to check. It doesn't look like this PR fixes the issue. I think it's because the `#line` mapping simply maps the whole using directive, it cannot map just a part of it as it's specified.